### DISCUSSION TITLE
Native format on save not working

### DISCUSSION THREAD

**Discussion by Aceticia** [LunarVim/LunarVim#2412] [Q&A] [answered]
With null-ls, I can see linting errors from flake8 on my python files. However, even if I specified `lvim.format_on_save=true` in my config file, this doesn't seem to be working. Is there something I'm missing from the doc?

**Comment by abzcoding** [comment]
can you elaborate?
`format_on_save` is for formatting (which is separate from linting)
more info -> https://www.lunarvim.org/languages/#linting-formatting

**Comment by abzcoding** [comment, marked as answer]
if i understand it correctly, you mean why don't you get formatted text in native lsp (without null-ls)
that is because some lang servers such as pyright, doesn't support formatting

run `:LvimInfo` while having a buffer open, look for `document_formatting` inside `capabilities`
<img width="957" alt="Screen Shot 2022-04-07 at 3 46 37 AM" src="https://user-images.githubusercontent.com/10992695/162088939-1501eec4-ffe9-4888-8fe2-7a721715740c.png">

and others do, for example, `rust_analyzer`
<img width="1506" alt="Screen Shot 2022-04-07 at 3 49 10 AM" src="https://user-images.githubusercontent.com/10992695/162089300-d4ab997c-1792-4d88-afeb-a129e8a78f1e.png">
